include/FileBrowser.h

```diff
@@ -108,26 +108,36 @@ class FileBrowserTreeWidget : public QTreeWidget
 
 
 private:
+	//! Start a preview of a file item
+	void previewFileItem(FileItem* file);
+	//! If a preview is playing, stop it. Returns false if nothing was playing
+	void stopPreview();
+
+	void keyPressEvent( QKeyEvent * ke ) override;
+	void keyReleaseEvent( QKeyEvent * ke ) override;
+	void hideEvent( QHideEvent * he ) override;
```

Comment:
What's our policy on overrides class visibility? These are under `private` whereas Qt has them under `protected`. I don't think we put any event handlers under `private` in any class.

Reply:
It depends on the situation. The general rule is to give things as little visibility as possible, but of course there are exceptions. You shouldn't reduce the visibility of `public` functions - these are a part of the public interface of the base class, so if they're not a part of the public interface of your derived class, you're probably using inheritance badly. It's okay to override `protected` methods as `private` if you want to stop further derived classes from calling your overridden implementation.

In this case though, I think they should stay `protected`. The way you get default event handling in Qt is to call the base class's implementation of the handler, so making them `private` makes this class very difficult to derive from. If this is the intention (although I doubt it), they should probably be declared `final` - it doesn't really make sense to allow a derived class to handle events, but force it to handle each event fully itself, and never pass it back up the hierarchy for default handling.

Reply:
Good point, fixed in latest commit!